src/services/algorithms_init/AlgorithmsInit_service.h

```diff
@@ -50,12 +50,12 @@ class AlgorithmsInit_service : public JService
             logger.init([this](const algorithms::LogLevel l, std::string_view caller, std::string_view msg){
                 static std::mutex m;
                 std::lock_guard<std::mutex> lock(m);
-                static std::map<std::string_view, std::shared_ptr<spdlog::logger>> loggers;
-                if (! loggers.contains(caller)) {
+                static std::map<std::string, std::shared_ptr<spdlog::logger>> loggers;
```

Comment:
Add comment.
```suggestion
                // storing the string_view is unsafe since it can become invalid
                static std::map<std::string, std::shared_ptr<spdlog::logger>> loggers;
```